flink-runtime/src/main/java/org/apache/flink/runtime/state/StateSnapshotContextSynchronousImpl.java

```diff
@@ -127,4 +128,38 @@ public OperatorStateCheckpointOutputStream getRawOperatorStateOutput() throws Ex
 		return new DoneFuture<>(stream.closeAndGetHandle());
 	}
 
-}
\ No newline at end of file
+	private <T extends StreamStateHandle> void closeAndUnregisterStream(NonClosingCheckpointOutputStream<T> stream) throws IOException {
+		Preconditions.checkNotNull(stream);
+
+		closableRegistry.unregisterClosable(stream.getDelegate());
+		stream.getDelegate().close();
+	}
+
+	public void close() throws IOException {
```

Comment:
I would prefer `AutoCloseable` over `Closeable`, because the former is general while the latter is more related to IO.